documentation/src/main/asciidoc/migration/index.asciidoc

```diff
@@ -43,16 +43,20 @@ Hibernate Search's requirements did not change in version {hibernateSearchVersio
 [[data-format]]
 == Data format and schema changes
 
-Indexes created with Hibernate Search {hibernateSearchPreviousStableVersionShort}
+Elasticsearch indexes created with Hibernate Search {hibernateSearchPreviousStableVersionShort}
 can be read from and written to with Hibernate Search {hibernateSearchVersion}.
 
+Reading and writing to Lucene indexes created with Hibernate Search {hibernateSearchPreviousStableVersionShort}
+using Hibernate Search {hibernateSearchVersion} may lead to exceptions, since there were incompatible changes applied to internal fields.
+It is recommended to recreate your Lucene indexes and reindex your database. The easiest way to do so is to use link:{hibernateSearchDocUrl}#indexing-massindexer[the `MassIndexer`] with link:{hibernateSearchDocUrl}#indexing-massindexer-parameters-drop-and-create-schema[`dropAndCreateSchemaOnStart(true)`].
```

Comment:
Actually you can be more forceful here since entity loading -- the default and probably only projection used by the vast majority of users -- requires to retrieve the identifier from the index.

```suggestion
You must recreate your Lucene indexes and reindex your database. The easiest way to do so is to use link:{hibernateSearchDocUrl}#indexing-massindexer[the `MassIndexer`] with link:{hibernateSearchDocUrl}#indexing-massindexer-parameters-drop-and-create-schema[`dropAndCreateSchemaOnStart(true)`].
```